### ISSUE TITLE
[a11y]: Overflow Menu -JAWS does not announce the Expanded/Collapsed State of the Menu

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16532]
### Package

@carbon/react

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

- The screen reader fails to announce the expand/collapse status when the user interacts with the menu.
- Upon hitting the spacebar, the options menu expands, but the screen reader does not announce it as "expanded."
- Conversely, when the user hits the spacebar again to collapse the menu, the screen reader incorrectly announces it as "expanded."

Collapsed state 
<img width="459" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/fb144d5c-0b68-49e1-a621-9b80c00e8c8c">

Expanded state 
<img width="436" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/5e5baba9-302a-46a4-af35-041eec8bcae4">

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
The issue was retested by our accessibility team (in @carbon/react v1.84), and the issue has been fixed.
